### ISSUE TITLE
Importing both named and default from same file not working.

### ISSUE THREAD

**Issue by frontendmonster** [babel/babel#6915]
bug report:
after upgrading to v7.0, ES6 import seems weird.
importing both named and default from file seems not working right.
named imports become undefined when adding default import on same file.

### Input Code
```js
import mongoose, {Schema} from 'mongoose';
console.log(Schema); // undefined
```
but it works when removing default import.
```js
import {Schema} from 'mongoose';
console.log(Schema) // object
```

### Babel/Babylon Configuration (.babelrc, package.json, cli command)
```js
{"presets": ["@babel/preset-env"] }
```

### Expected Behavior
```js
import mongoose from 'mongoose';
import {Schema} from 'mongoose';
console.log(Schema); // object
```

### Current Behavior
```js
import mongoose from 'mongoose';
console.log(Schema); // undefiend
```
### Your Environment

| software         | version(s)
| ---------------- | -------
| Babel            |  7.0.0-beta.32
| node             | 9.2.0
| npm              | 5.5.1
| Operating System | Win10-x64

**Comment by nicolo-ribaudo** [comment]
mongoose only has a default export, not named exports (which are the equivalent of doing `module.exports.Schema = Schema`):
https://github.com/Automattic/mongoose/blob/master/lib/index.js#L860

**Comment by Ay-zA** [comment]
@nicolo-ribaudo Ok, but why `` import { Schema } from 'mongoose'; `` works?

**Comment by frontendmonster** [comment]
@nicolo-ribaudo Ok, but why ``import { Schema } from 'mongoose';`` works?

**Comment by loganfsmyth** [comment]
Hmm, this is a pain. The `mongoose` module puts `.Schema` on the prototype of the `module.exports` object.

So if you only have a named export, we don't bother using our `interopRequireWildcard` helper, but if you include the default, we do. The helper ignores prototype properties on `module.exports`, but the non-helper cases does access them.

We can leave this open so we can track it, but my recommendation would absolutely be to do
```
import mongoose from 'mongoose';
console.log(mongoose.Schema);
```
when you need the schema. ES6 import statements are not destructuring statements. This behavior may yet change more for Babel 7, we haven't fully decided yet.

**Comment by nicolo-ribaudo** [comment]
I'm tempted to mark this as `won't fix`/`by design`, since `module.exports = new Mongoose`, which mongoose uses [1], is the equivalent of `export default new Mongoose`. Thus, `import { Schema } from "mongoose"` is wrong.

cc @vkarpov15 (I'm pinging you because by looking at mongoose's commits you are the main maintainer).
What do you think about this?

[1] https://github.com/Automattic/mongoose/blob/6da0c525f5ab080a78631278cdb0d7d435ca976d/lib/index.js#L863

**Comment by vkarpov15** [comment]
@nicolo-ribaudo I agree, statements like `import * from 'mongoose'` and `import {model} from 'mongoose'` do not work with mongoose because of how the lib is structured. I opened up an issue on the mongoose repo (https://github.com/Automattic/mongoose/issues/5940) to make a note of this on the readme.
